bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -104,9 +106,24 @@ public void initialize() {
                             insteonBinding.logDeviceStatistics();
                         }, 0, LOG_DEVICE_STATISTICS_DELAY_IN_SECONDS, TimeUnit.SECONDS);
 
-                        insteonBinding.setIsActive(true);
+                        // wait until driver is initialized before setting network to ONLINE
+                        driverInitializedJob = scheduler.scheduleWithFixedDelay(() -> {
+                            logger.debug("driver is not initialized yet");
```

Comment:
This log could be wrong.
You should move it in the else branch of your next if.
Or this is the message which is not appropriate.

Reply:
moved it